go/marshal/encode_test.go

```diff
@@ -694,6 +694,62 @@ func TestEncodeOriginal(t *testing.T) {
 	assert.True(out.Type().Equals(st))
 }
 
+type MarshalOptionsTestT struct {
+	Abc  int
+	opts MarshalOptions `noms:"-"`
+}
+
+func (t MarshalOptionsTestT) GetMarshalOptions() MarshalOptions {
+	return t.opts
+}
+
+func TestEncodeMarshalOptionsName(t *testing.T) {
+	assert := assert.New(t)
+
+	s := MarshalOptionsTestT{
+		Abc: 42,
+		opts: MarshalOptions{
+			Name: "",
+		},
+	}
+	v, err := Marshal(s)
+	assert.NoError(err)
+	assert.True(types.NewStruct("MarshalOptionsTestT", types.StructData{
```

Comment:
It's hard with default Go values for structs. It would mean that any type that implements the MarshalOptionsGetter, for any reason, would change its behavior.